### PR TITLE
Feature/os 88 refactor service layer fe

### PR DESCRIPTION
**Pull request by Avedis92**

### CODE REVIEW

File: os-ui/src/pages/Homepage/Homepage.module.scss [hunk 1]
```diff
@@ -2,7 +2,7 @@
 @import "../../styles/Mixins/FlexMixins.scss";
 @import "../../styles/Mixins/FontMixin.scss";
 
-.categoriesMainContainer {
+/*.categoriesMainContainer {
```
Comment (SevakMart) [inline os-ui/src/pages/Homepage/Homepage.module.scss:5]: if we don't need this, just remove

File: os-ui/src/pages/Homepage/Subcomponents/Mentors/Mentors.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            handleArrowClick={() => {
+              setPage((prevPage) => prevPage - 1);
+            }}
+          />
+        ) : null}
+        {
+        mentors.length > 0 && !errorMessage ? mentors.map((mentor, index) => (
+          <MentorCard
+            key={index}
+            name={mentor.name}
```
Comment (SevakMart) [inline os-ui/src/pages/Homepage/Subcomponents/Mentors/Mentors.tsx:52]: just pass mentor to the mentor card, then inside it use its properties

File: os-ui/src/services/fetchData.tsx [hunk 3]
```diff
@@ -1,21 +1,18 @@
...
-    },
-  }).catch(handleError));
-  const data = await response.json();
-  return data;
-};
+  async get(path: string, params: object = {}, token = '') {
+    return request(`${this.baseUrl}/${path}`, 'GET', params, token);
+  }
+
+  async post(path: string, body:unknown, params: object, token = '') {
```
Comment (SevakMart) [inline os-ui/src/services/fetchData.tsx:14]: can't we make the body an object?

File: os-ui/src/types/SearchType.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
 export interface SearchProps {
-    pathname:string;
-    searchKeyName:string;
+    /* pathname:string;
```
Comment (SevakMart) [inline os-ui/src/types/SearchType.ts:2]: remove comments if it is redundant

**Review by SevakMart — CHANGES_REQUESTED** [review]
please fix comments

File: os-ui/src/pages/Homepage/Homepage.module.scss (hunk 1 above)
Comment (Avedis92) [reply os-ui/src/pages/Homepage/Homepage.module.scss]: Done

File: os-ui/src/pages/Homepage/Subcomponents/Mentors/Mentors.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  const [page, setPage] = useState(0);
+  const [maxPage, setMaxPage] = useState(10);
+  const [errorMessage, setErrorMessage] = useState('');
+  const { mentorMainContainer, mentorListContainer } = styles;
+
+  useEffect(() => {
+    const cancel = false;
+    let mentorPromise;
+
+    if (isLoggedIn)mentorPromise = userService.getMentors({ page, size: 4 });
```
[the remark this answers is not in the document]
Comment (Avedis92) [reply os-ui/src/pages/Homepage/Subcomponents/Mentors/Mentors.tsx:23]: I will refactor it on coming days. Please create a task for it.

File: os-ui/src/pages/Homepage/Subcomponents/Mentors/Mentors.tsx (hunk 2 above)
Comment (Avedis92) [reply os-ui/src/pages/Homepage/Subcomponents/Mentors/Mentors.tsx]: Done

File: os-ui/src/services/fetchData.tsx (hunk 3 above)
Comment (Avedis92) [reply os-ui/src/services/fetchData.tsx:14]: Still waiting for Hakob to send me his updates about it

File: os-ui/src/types/SearchType.ts (hunk 4 above)
Comment (Avedis92) [reply os-ui/src/types/SearchType.ts]: Done

**Review by SevakMart — APPROVED** [review]

**Review by SevakMart — APPROVED** [review]

**Pull request merged**